### PR TITLE
(Feature) Allow PoA contract to handle hard forks

### PR DESCRIPTION
**Pull request by rstormsf**
In order to use multiset feature from parity, we have to be able to
1. Set ProxyStorage via constructor
2. Set Previous Set of Validators

### CODE REVIEW

**Comment by wadeAlexC** [comment]
Looks good to me!

File: contracts/PoaNetworkConsensus.sol [hunk 1]
```diff
@@ -62,12 +62,15 @@ contract PoaNetworkConsensus is IPoaNetworkConsensus {
...
 
-    function PoaNetworkConsensus(address _masterOfCeremony) public {
+    function PoaNetworkConsensus(address _masterOfCeremony, address[] validators) public {
         // TODO: When you deploy this contract, make sure you hardcode items below
         // Make sure you have those addresses defined in spec.json
         require(_masterOfCeremony != address(0));
         masterOfCeremony = _masterOfCeremony;
         currentValidators = [masterOfCeremony];
+        for (uint256 y = 0; y < validators.length; y++) {
+            currentValidators.push(validators[y]);
```
Comment (vbaranov) [inline contracts/PoaNetworkConsensus.sol]: We don't check here, that `validators[y] != masterOfCeremony`. Should we?

Comment (rstormsf) [reply contracts/PoaNetworkConsensus.sol]: MoC is already added to list of validators, we assume that people should not double include a validator. Worse case, they will re deploy the contract if it is deployed incorrectly

Comment (vbaranov) [reply contracts/PoaNetworkConsensus.sol]: Ok, seems it is ready to merge in this case.

**Pull request merged**